stage1/init/common/pod.go

```diff
@@ -449,6 +449,40 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		unit.NewUnitOption("Service", "CapabilityBoundingSet", strings.Join(capabilitiesStr, " ")),
 	}
 
+	if ra.ReadOnlyRootFS {
+		opts = append(opts, unit.NewUnitOption("Service", "ReadOnlyDirectories", "/"))
+	}
+
+	// TODO(tmrts): Extract this logic into a utility function.
+	vols := make(map[types.ACName]types.Volume)
+	for _, v := range p.Manifest.Volumes {
+		vols[v.Name] = v
+	}
+
+	rwDirs := []string{}
+	imageManifest := p.Images[appName.String()]
+	for _, m := range GenerateMounts(ra, vols, imageManifest) {
+		vol := vols[m.Volume]
+
+		absRoot, err := filepath.Abs(p.Root) // Absolute path to the pod's rootfs.
+		if err != nil {
+			return errwrap.Wrap(errors.New("could not get pod's root absolute path"), err)
+		}
+
+		appRootfs := common.AppRootfsPath(absRoot, appName)
+
+		mntPath, err := EvaluateSymlinksInsideApp(appRootfs, m.Path)
+		if err != nil {
+			continue
```

Comment:
Do you ignore the error because this case should have been caught previously? Difficult to see if it is correct.